api/base/serializers.py

```diff
@@ -769,6 +775,25 @@ def invalid_embeds(self, fields, embeds):
         invalid_embeds = set(embeds.keys()) - set([f.field_name for f in fields_check if getattr(f, 'json_api_link', False)])
         return invalid_embeds
 
+    def to_esi_representation(self, data):
+        href = None
+        query_params_blacklist = ['page[size]', 'format']
+        try:
+            href = data.get_absolute_url()
+        except:
+            representation = super(JSONAPISerializer, self).to_representation(data)
+            return representation
+        else:
+            if href and not href == '{}':
```

Comment:
So, if get_absolute_url succeeds but sends back {} or None, there's nothing that gets returned. Is that the proper case? I think that would return a None into the list. Which might serialize fine, I'm not sure.